lib/Driver/DarwinToolChains.cpp

```diff
@@ -855,6 +860,13 @@ toolchains::Darwin::validateArguments(DiagnosticEngine &diags,
   if (args.hasArg(options::OPT_static_stdlib)) {
     diags.diagnose(SourceLoc(), diag::error_darwin_static_stdlib_not_supported);
   }
+
+  // If a C++ standard library is specified, it has to be libc++.
+  if (auto arg = args.getLastArg(options::OPT_experimental_cxx_stdlib)) {
+    if (StringRef(arg->getValue()) != "libc++") {
```

Comment:
Not sure what you mean? I don't have a `std::string` here yet -- `arg->getValue()` returns a `const char *`. Creating a string from that would require a memory allocation -- using a `StringRef` instead avoids that.

Reply:
For some reason I thought that there was `int (const char *, const char *)` overload that allowed you to safely get to `strncmp`.  I just dislike the `StringRef` construction since it tends to be a bit tricky, but sure.

Reply:
Fine to mark this resolved then?